client/src/client.ts

```diff
@@ -69,22 +81,57 @@ export class AngularLanguageClient implements vscode.Disposable {
         provideHover: async (
             document: vscode.TextDocument, position: vscode.Position,
             token: vscode.CancellationToken, next: lsp.ProvideHoverSignature) => {
-          if (isInsideInlineTemplateRegion(document, position)) {
-            return next(document, position, token);
+          if (!isInsideInlineTemplateRegion(document, position)) {
+            return;
+          }
+
+          const angularResults = await next(document, position, token);
+          // If Angular provided results, do not perform request forwarding.
+          if (angularResults) {
+            return angularResults;
           }
+
+          const vdocUri = this.createVirtualHtmlDocForPosition(document, position);
+          const result = await vscode.commands.executeCommand<vscode.Hover[]>(
+              'vscode.executeHoverProvider', vdocUri, position);
+          return result?.[0];
         },
         provideCompletionItem: async (
             document: vscode.TextDocument, position: vscode.Position,
             context: vscode.CompletionContext, token: vscode.CancellationToken,
             next: lsp.ProvideCompletionItemsSignature) => {
-          if (isInsideInlineTemplateRegion(document, position)) {
-            return next(document, position, context, token);
+          // If not in inline template, do not perform request forwarding
+          if (!isInsideInlineTemplateRegion(document, position)) {
+            return;
           }
+          const angularCompletions =
+              await next(document, position, context, token) as vscode.CompletionItem[] | null |
+              undefined;
+
+          const vdocUri = this.createVirtualHtmlDocForPosition(document, position);
+          // This will not include angular stuff because the vdoc is not associated with an angular
+          // component
+          const htmlProviderCompletions =
+              await vscode.commands.executeCommand<vscode.CompletionList>(
+                  'vscode.executeCompletionItemProvider', vdocUri, position,
+                  context.triggerCharacter);
+          return [...(angularCompletions ?? []), ...(htmlProviderCompletions?.items ?? [])];
         }
       }
     };
   }
 
+  private createVirtualHtmlDocForPosition(document: vscode.TextDocument, position: vscode.Position):
+      vscode.Uri {
+    const originalUri = document.uri.toString();
+    const vdocContents = getHTMLVirtualContent(document, position);
+    this.virtualDocumentContents.set(originalUri, vdocContents);
+
+    const vdocUriString = `angular-embedded-content://html/${encodeURIComponent(originalUri)}.html`;
```

Comment:
Consider using the `Uri.file()` method, then call `.with()` to change the scheme.